tests/phpunit/tests/term/getTerms.php

```diff
@@ -2988,6 +2988,191 @@ public function test_with_term_slug_equal_to_string_0() {
 		$this->assertSameSets( array( $term_id ), wp_list_pluck( $found, 'term_id' ) );
 	}
 
+
+	/**
+	 * @ticket 55352
+	 *
+	 * @dataProvider data_same_term_args
+	 */
+	public function test_cache_key_generation( $args_1, $args_2 ) {
+		global $wpdb;
+		register_taxonomy( 'wptests_tax', 'post', array( 'hierarchical' => true ) );
+
+		$term_id1 = self::factory()->term->create(
+			array(
+				'name'     => 'Foo',
+				'slug'     => 'Foo',
+				'taxonomy' => 'wptests_tax',
+			)
+		);
+
+		$term_id2 = self::factory()->term->create(
+			array(
+				'name'     => 'Bar',
+				'slug'     => 'bar',
+				'taxonomy' => 'wptests_tax',
+			)
+		);
+
+		$posts = self::factory()->post->create_many( 3, array( 'post_type' => 'post' ) );
+		foreach ( $posts as $i => $post ) {
+			wp_set_object_terms( $post, array( $term_id1, $term_id2 ), 'wptests_tax' );
+		}
+
+		$query1      = get_terms( $args_1 );
+		$num_queries = $wpdb->num_queries;
+		$query2      = get_terms( $args_2 );
+
+		$this->assertSame( $num_queries, $wpdb->num_queries );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_same_term_args() {
+		return array(
+			array(
```

Comment:
```suggestion
			'all fields vs ids only' => array(
```

* :1234: phpunit displays the key to identify which dataset item is used, naming them makes it so much clearer when there is a bug. 
* I'd also add a test case where everything is identical except for the order, ie a-z for query one, z-a for query two.

Reply:
Fixed.